librepo/downloader.c

```diff
@@ -489,7 +489,7 @@ lr_headercb(void *ptr, size_t size, size_t nmemb, void *userdata)
     }
 
     #ifdef WITH_ZCHUNK
-    if(lrtarget->target->is_zchunk && lrtarget->mirror->max_ranges > 0)
+    if(lrtarget->target->is_zchunk && lrtarget->mirror->max_ranges > 0 && lrtarget->mirror->mirror->protocol != LR_PROTOCOL_FTP)
```

Comment:
I didn't investigate it thoroughly so I might be wrong. Zchunk is usable only for protocols supporting ranges - HTTP(S) at the moment. Shouldn't be this test positive - i.e. `if (... && does_protocol_support_ranges(lrtarget->mirror->mirror->protocol))` (or even `&& protocol == LR_PROTOCOL_HTTP`). I'm basically thinking about other protocols librepo can handle (like rsync) which are not fixed by this patch.

Reply:
Zchunk does only work with http and https.  This would probably be better with just `&& protocol == LR_PROTOCOL_HTTP` (assuming that also covers https).

Reply:
Ok I have switched all the conditions to not be FTP specific but rather HTTP specific. (LR_PROTOCOL_HTTP also covers https)